Dockerfile

```diff
@@ -24,38 +24,53 @@ RUN apk add --no-cache \
   tzdata \
   openssl \
   postgresql-dev \
-  libc6-compat
-
-RUN bundle install --jobs 4 --retry 5 && \
-  chmod -R a+r /usr/local/bundle
+  libc6-compat && \
+  bundle config --global without "${BUNDLE_WITHOUT}"  && \
+  bundle config --global path "${BUNDLE_PATH}" && \
+  bundle config --global deployment "${BUNDLE_DEPLOYMENT}" && \
+  bundle config --global retry 5 && \
+  bundle install && \
+  find /usr/local/bundle/ \
+    \( \
+      -name "*.c" -o \
+      -name "*.o" -o \
+      -name "*.a" -o \
+      -name "*.h" -o \
+      -name "Makefile" -o \
+      -name "*.md" \
+    \) -delete && \
+  chmod -R a+r "${BUNDLE_PATH}"
 
 # Final stage
 FROM base
 LABEL maintainer="keygen.sh <oss@keygen.sh>"
 
-WORKDIR /app
-COPY . /app
-
 RUN apk add --no-cache \
   bash \
   postgresql-client \
   tzdata \
   libc6-compat
 
+COPY --from=build --chown=keygen:keygen \
+  /usr/local/bundle/ /usr/local/bundle
+
+WORKDIR /app
+COPY . /app
+
 ENV KEYGEN_EDITION="CE" \
     KEYGEN_MODE="singleplayer" \
     RAILS_LOG_TO_STDOUT="1" \
     PORT="3000" \
     BIND="0.0.0.0"
 
-RUN chmod +x /app/scripts/entrypoint.sh && \
+RUN \
+  chmod +x /app/scripts/entrypoint.sh && \
   adduser -h /app -g keygen -u 1000 -s /bin/bash -D keygen && \
   chown -R keygen:keygen /app
 
-COPY --from=build --chown=keygen:keygen \
-  /usr/local/bundle/ /usr/local/bundle
-
 USER keygen
 
+EXPOSE $PORT
```

Comment:
The port is only relevant for the web process, which is why it wasn't here before. For example, the worker process should not be exposed on a port.

Reply:
That makes sense. However, there's no downside to exposing it for other cases. It will be simply not used.

Reply:
Reverted this any way to unblock this:
[`01ec10e` (#813)](https://github.com/keygen-sh/keygen-api/pull/813/commits/01ec10ed3bfbd7455117e27a5a0e54f6b168a158)

But we can discuss bringing it back.